platform/mock-radio.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Mock radio for interfacing with mock devices
+ */
+
+export interface MockUDPListener {
+  onUDPMessage(msg: Buffer, rinfo: RemoteAddressInfo): void;
+}
+
+export class RemoteAddressInfo {
+  address: string;
+  family: string;
+  port: number;
+  size: number;
+  constructor(address: string, family: string, port: number, size: number) {
+    this.address = address;
+    this.family = family;
+    this.port = port;
+    this.size = size;
+  }
+}
+
+// Simulates a network with simple UDP messaging functionality
+export class MockNetwork {
+  udpListeners: Map<string, MockUDPListener[]>;
+
+  constructor() {
+    this.udpListeners = new Map<string, MockUDPListener[]>();
+  }
+
+  registerUDPListener(
```

Comment:
could this be using a `EventEmitter` rather than implementing your own listener pattern?
https://nodejs.org/api/events.html

Reply:
I hadn't seen this.  Opened issue #26 for this